docs/reference/components/sinks/redis.cue

```diff
@@ -32,7 +32,7 @@ components: sinks: redis: {
 			}
 			request: {
 				enabled:                    true
-				concurrency:                5
+				concurrency:                1
```

Comment:
I believe this will render this field as configurable in the docs, but I see we hardcoded it to 1 below. I think we should default to 1 but still let users modify it if they don't care about in-order messages.

Theoretically we could allow concurrent requests with in-order messages if we partitioned the requests on the key they are writing to. I don't think we have an existing example of doing this in Vector though so it'd be some amount of work. I'm fine with just defaulting to 1 for now and we can follow up later to improve performance if needed.

Reply:
@jszwedko 
> I believe this will render this field as configurable in the docs, but I see we hardcoded it to 1 below. I think we should default to 1 but still let users modify it if they don't care about in-order messages.

Do you mean this code?

```rust
concurrency: Concurrency::Fixed(1),
```
This should be just a default value，it doesn't seem to be hardcoded，and it seems that we can still configure it through the `request` option

> Theoretically we could allow concurrent requests with in-order messages if we partitioned the requests on the key they are writing to. I don't think we have an existing example of doing this in Vector though so it'd be some amount of work. I'm fine with just defaulting to 1 for now and we can follow up later to improve performance if needed.

partitioned the requests, this is cool and it will be very similar to kafka

Reply:
Ah yes, I see. I missed that this could still be overridden. 👍